### PR TITLE
[Blogpost] Twitter Digest

### PR DESCRIPTION
**Pull request by Lily2point0**

### CODE REVIEW

**Comment by railsagainstignorance** [comment]
would be worth mentioning that this is a self-installed chrome extension, and not sitting in the chrome store.

Also, it feels like another line or two would adding about the benefits of this grouping of tweets into two distinct sets.
- Pulling out all the tweets matching the criteria creates a condensed view of of those matching tweets, and a less noisy view of the non-matching tweets, either (or perhaps both) of which are likely to be of benefit to the reader.

File: _posts/2017-06-23-twitter-digest.html [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+<p>
+	Any tweet in the user's timeline containing the keyword is then 'extracted' from it and appears in a cluster at the top of the timeline. It is then up to them to choose to explore the tweets in the cluster or scroll past them.
+</p>
+<p>
+	Twitter Digest is merely an attempt to reduce the scattered noise in a feed by condensing and surfacing tweets. It differs from other filters or "spoiler alert" type of extensions as it doesn't hide or dismiss any content but offers a new reading mode.
+</p>
+
+<img src="/assets/uploads/2017/06/twitter_digest.png" alt="Twitter Digest in situ." />
+
+<h2>Technical details & challenges</h2>
```
Comment (seanmtracey) [inline _posts/2017-06-23-twitter-digest.html]: `&amp;` (or does Jekyll do that for us?)

File: _posts/2017-06-23-twitter-digest.html [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+<p>
+	Twitter Digest comprises of a node server, which handles authentication and keyword requests, and a Chrome extension which interacts with the client side (in this case the DOM of <a href="https://twitter.com/" target="_blank">twitter.com</a>).
```
Comment (seanmtracey) [inline _posts/2017-06-23-twitter-digest.html]: 'node' > 'Node'

File: _posts/2017-06-23-twitter-digest.html [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+<p>
+	The source code is available on <a href="https://github.com/ftlabs/Twitter-digest" target="_blank">GitHub</a>.
+</p>
+<p>
+	In the early stages of the prototype, Twitter Digest didn't allow for a choice of keyword and required a JSON object to be pasted in the popup of the extension to be processed. It can now handle switching between accounts, signing in with Twitter and toggling the digest on and off. We have made a few discoveries throughout the process.
+</p>
+
+	<h3>Twitter API</h3>
+<p>
+	It took a few tries to find the correct endpoint of the Twitter API for the digest. After considering <a href="https://dev.twitter.com/rest/reference/get/statuses/user_timeline" target="_blank">user_timeline</a>, we settled on <a href="https://dev.twitter.com/rest/reference/get/statuses/home_timeline" target="_blank">home_timeline</a> which offers the most accurate depiction of what is in a user's feed, although it excludes tweets followees have liked and the 'In case you missed it section'.
```
Comment (seanmtracey) [inline _posts/2017-06-23-twitter-digest.html]: What is a 'followee'?

File: _posts/2017-06-23-twitter-digest.html [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+</p>
+<p>
+	OAuth, as always, has proved tricky, especially with the added layer of communication between the extension and the Node server. Ensure you have all the necessary callbacks in place, and above all that the callback URL set when creating the Twitter application matches the one you've implemented. (This needs to be amended when switching from a local environment to a live server).
```
Comment (seanmtracey) [inline _posts/2017-06-23-twitter-digest.html]: This is a weird sentence from "Ensure" onwards. It's as if we're walking a developer through making their own Twitter client, which I don't think fits well in this context.

File: _posts/2017-06-23-twitter-digest.html [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+</p>
+	<h3>Chrome extension</h3>
+<p>
+	The manifest file is crucial for a Chrome extension to work. You should reference all the properties and domains that are going to be required in there (especially to avoid CORS issues).
```
Comment (seanmtracey) [inline _posts/2017-06-23-twitter-digest.html]: Following on from my last comment, this whole section feels like more of a walkthrough than discussion of technical limitations/challenges/design choices that we came across. I don't think it fits well within the context of the post.

**Comment by railsagainstignorance** [comment]
LGTM. Deploy when ready.

**Pull request merged**